src/core/security/credentials.c

```diff
@@ -1213,32 +1093,46 @@ grpc_credentials *grpc_credentials_contains_type(
   return NULL;
 }
 
+grpc_channel_credentials *grpc_composite_channel_credentials_create(
+    grpc_channel_credentials *channel_creds, grpc_call_credentials *call_creds,
+    void *reserved) {
+  GPR_ASSERT(reserved == NULL);
+  if (channel_creds == NULL) return NULL;
+  if (channel_creds->call_creds == NULL) {
+    channel_creds->call_creds = grpc_call_credentials_ref(call_creds);
+  } else if (call_creds != NULL) {
+    grpc_call_credentials *composite_creds =
+        grpc_composite_call_credentials_create(channel_creds->call_creds,
+                                               call_creds, NULL);
+    if (composite_creds == NULL) return NULL;
+    grpc_call_credentials_unref(channel_creds->call_creds);
+    channel_creds->call_creds = composite_creds;
```

Comment:
Why does this modify the channel credentials in place? I very much expect composition to be idempotent.